### PR TITLE
Fix radio scanner runtime when speaker is not a mob

### PR DESCRIPTION
**Pull request by Sovexe**
## About the PR 
proc now takes an atom movable and only checks for a voice changer if you're a human

## Why's this needed? 
File | MechanicMadness.dm
-- | --
Line | 2424
Error | undefined variable /obj/machinery/computer/announcement/var/wear_mask
```
proc name: hear radio (/obj/item/mechanics/radioscanner/proc/hear_radio)
  source file: MechanicMadness.dm,2424
  usr: null
  src: Radio Scanner Component (/obj/item/mechanics/radioscanner)
  src.loc: the steel floor (136,188,1) (/turf/simulated/floor)
  call stack:
Radio Scanner Component (/obj/item/mechanics/radioscanner): hear radio(Bridge Announcement Computer (/obj/machinery/computer/announcement), /list (/list), "english")
Station Intercom (Radio) (/obj/item/device/radio/intercom): talk into(Bridge Announcement Computer (/obj/machinery/computer/announcement), /list (/list), 0, "Bridge Announcement Computer", "english")
Bridge Announcement Computer (/obj/machinery/computer/announcement): announce arrival("Sov Extant", "Captain")
the industrial cryogenics unit (/obj/cryotron): spawn next person()
```

### CODE REVIEW

File: code/WorkInProgress/MechanicMadness.dm [hunk 1]
```diff
@@ -2413,16 +2413,22 @@
 		frequency = new_frequency
 		radio_connection = radio_controller.add_object(src, "[frequency]")
 		tooltip_rebuild = 1
-	proc/hear_radio(mob/M as mob, msg, lang_id)
+	proc/hear_radio(atom/movable/AM as mob|obj, msg, lang_id)
```
Comment (ZeWaka) [inline code/WorkInProgress/MechanicMadness.dm:2416]: Unless this is used as a verb, the 'as X' is useless (and might decrease performance?) iirc

File: code/WorkInProgress/MechanicMadness.dm [hunk 2]
```diff
@@ -2413,16 +2413,22 @@
...
 		if (level == 2) return
 		LIGHT_UP_HOUSING
 		var/message = msg[2]
 		if (lang_id in list("english", ""))
 			message = msg[1]
 		message = strip_html(html_decode(message))
-		var/heardname = M.real_name
```
Comment (ZeWaka) [inline code/WorkInProgress/MechanicMadness.dm:2423]: Why would you put a : in the replacement of this? It'll runtime if you pass an obj...

Comment (Sovexe) [reply code/WorkInProgress/MechanicMadness.dm:2423]: `obj` has `real_name` as well

File: code/WorkInProgress/MechanicMadness.dm (hunk 1 above)
Comment (Sovexe) [reply code/WorkInProgress/MechanicMadness.dm]: removed

File: code/WorkInProgress/MechanicMadness.dm (hunk 2 above)
Comment (Sovexe) [reply code/WorkInProgress/MechanicMadness.dm:2423]: actually I'm going to change this anyway since `name` makes more sense for objects

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**